### PR TITLE
chore(contracts): defaults lib w/ address, unit256, xdomain msg sender

### LINKED ISSUE #803: Make consistent default addresses
We should use the same default values as currently in the Execution manager we use `0xdEfa017defA017DeFA017DEfa017DeFA017DeFa0` to signify a default address and in the Cross domain messengers we use `0x000000000000000000000000000000000000dEaD` for the default sender.

It would also be nice to move this (as well as the DEFAULT_UINT256 from the Execution Manager) into a small library.

Logged as a result of https://github.com/ethereum-optimism/optimism/pull/667#discussion_r627532765

### PR DESCRIPTION
**Pull request by davisshaver**
**Description**
Adds default values library for default address, uint256, & xdomain msg sender.

**Additional context**
n/a

**Metadata**
- Fixes #803

### CODE REVIEW

File: packages/contracts/contracts/optimistic-ethereum/libraries/utils/Lib_DefaultValues.sol
Comment (smartcontracts) [inline packages/contracts/contracts/optimistic-ethereum/libraries/utils/Lib_DefaultValues.sol:10]: Address here should be `0xdEfa017defA017DeFA017DEfa017DeFA017DeFa0`

File: packages/contracts/contracts/optimistic-ethereum/libraries/utils/Lib_DefaultValues.sol
Comment (smartcontracts) [inline packages/contracts/contracts/optimistic-ethereum/libraries/utils/Lib_DefaultValues.sol:1]: Could you actually move this file into the `/constants` folder?

File: packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol
Comment (smartcontracts) [inline packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol:41]: IMO instead of using `is Lib_DefaultValues` it would be better to do `Lib_DefaultValues.DEFAULT_ADDRESS`. It's a little more explicit.

**Review by smartcontracts — CHANGES_REQUESTED** [review]
Thanks for the PR! Requesting a few minor changes but glad someone is taking up this issue :-)

**Review by smartcontracts — APPROVED** [review]
LGTM! Thank you!

**Comment by tynes** [comment]
Are we sure that this is a non breaking change in the sense the bytecode will be 1:1? If no, this needs to target the experimental branch

**Comment by smartcontracts** [comment]
Actually I need to dismiss my original approval. Since this PR changes contracts it is considered a breaking change and must therefore target the latest release branch (the `experimental` branch). However, these constant values may no longer exist within the `experimental` branch. I would recommend taking a look at that branch and checking if this PR would still apply there. If so, feel free to retarget that branch.

**Pull request closed without merging**
